src/lib/storage/abstract_table_constraint.cpp

```diff
@@ -30,3 +29,12 @@ bool AbstractTableConstraint::operator!=(const AbstractTableConstraint& rhs) con
 }
 
 }  // namespace opossum
+
+namespace std {
+
+size_t hash<opossum::AbstractTableConstraint>::operator()(
+    const opossum::AbstractTableConstraint& table_constraint) const {
+  return table_constraint.hash();
+}
```

Comment:
What is this required for?

Reply:
We might want to use hash-based containers, but I can also remove it

Reply:
I don't have a strong opinion on this, but I'm tending to remove it since we don't need it now and add it in the future if it's needed at some point.